exla/lib/exla.ex

```diff
@@ -184,6 +176,45 @@ defmodule EXLA do
 
   @behaviour Nx.Defn.Compiler
 
+  @doc """
+  Sets the global defn options to the EXLA compiler with the
+  preferred client based on the available platforms.
+
+  This function is typically invoked at the top of scripts
+  and code notebooks which might be potentially executed from
+  multiple platforms. Do not invoke this function during runtime,
+  as it changes the `Nx.Defn` options globally. If you have a
+  specific client that you want to use throughout your project,
+  use configuration files instead:
+
+      config :nx, :default_defn_options, [compiler: EXLA, client: :cuda]
+
+  ## Examples
+
+      EXLA.set_preferred_defn_options([:tpu, :cuda, :rocm, :host])
+
+  The above will try to find the first client available and set
+  the `EXLA` compiler with the client as the compilers for `Nx.Defn`.
+  If no client is found, `EXLA` is not set as compiler at all,
+  therefore it is common to add `:host` as the last option.
+  """
+  def set_preferred_defn_options(clients) do
```

Comment:
Would it make sense to also accept additional options like `:keep_on_device`?

Reply:
Generally speaking, we should be pushing towards using configuration files to change the application environment, which is why we have warnings in the docs to avoid calling the global functions unless scripting.